pkg/distributor/push.go

```diff
@@ -190,6 +190,30 @@ func handler(
 	})
 }
 
+// Retryable HTTP codes for otlp collector
+var retryableHTTPCodes = map[int]bool{
+	http.StatusBadGateway:         true,
+	http.StatusServiceUnavailable: true,
+	http.StatusGatewayTimeout:     true,
+	http.StatusTooManyRequests:    true,
+}
+
+// convertHTTPCodeOtlp maps specific HTTP status codes to those recognized by the OTLP client for retry.
+//   - Maps a custom 529 error to the standard 429 (Too Many Requests) error.
+//   - Converts non-retryable 5xx errors to 503 (Service Unavailable) to trigger client retries, as per OTLP specifications (https://opentelemetry.io/docs/specs/otlp/#failures-1).
+//   - Ensures that the 501 (Not Implemented) error is not retried.
+var convertHTTPCodeOtlp = func(httpCode int) int {
```

Comment:
[Nit] Is there any reason not to use regular non-variable function syntax?
```suggestion
func convertHTTPCodeOtlp(httpCode int) int {
```

Reply:
this was used for testing overwrite, since we decided to go for dublicate code instead, this is removed.